app/Http/Controllers/Auth/CaasLoginController.php

```
@@ -33,24 +33,44 @@ public function loginCheck(Request $request)
     }
 
 
-    public function changePass(Request $request,$id)
+    public function changePassword(Request $request)
     {
-        $this->validate($request, [
-            'password'  => 'required|min:8|string',
-            'confirmPassword'  => 'required|same:password',
-        ]);
-        DataCaas::where('id', $id)->update([
-            'password' => Hash::make($request->password),
-        ]);
-        Auth::guard('caas')->logout();
-        return redirect()->route('caas.login');
+        $id = Auth::id();
+        $caas = Datacaas::find($id);
+
+        // Validasi
+        $rules = [
+            'old_password' => 'required|password',
+            'password' => 'required|min:8|regex:/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/',
+        ];
+
+        $messages = [
+            'old_password.password' => 'Old password is incorrect.',
+            'password.regex' => 'Password must contain at least one uppercase letter, one lowercase letter, one number, and one special character.',
+        ];
+
+        $this->validate($request, $rules, $messages);
+
+        // Verifikasi password lama
+        if (!Hash::check($request->old_password, $caas->password)) {
+            return redirect()->back()->withErrors(['old_password' => 'Old password is incorrect.']);
```

Here, when we want to return the error, can we utilise variable `$messages`??

---

instead of creating an exact same thing there

---

okay bang i will do it

---

Please do it in the next PR okay,